membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -60,16 +63,35 @@ object AccountDetails {
         case _ => Json.obj()
       }
 
-      val alertText = membershipAlertText match {
-        case Some(text) => Json.obj("alertText" -> text)
-        case None => Json.obj()
-      }
+      def planIsCurrent(plan: SubscriptionPlan.AnyPlan) = (plan.start == now || plan.start.isBefore(now)) && (plan match {
+        case paidPlan: SubscriptionPlan.Paid => paidPlan.end.isAfter(now)
+        case _ => true
+      })
+
+      def jsonifyPlan(plan: SubscriptionPlan.AnyPlan) = Json.obj(
+        "productRatePlanId" -> plan.productRatePlanId.get, // consider exposing hash of this to avoid exposing internal IDs
```

Comment:
Generally I am not keen on the idea of exposing productRatePlanIds. This is Zuora jargon and it would be nice if we could abstract it away at this level so clients didn't need to care about such things!

Reply:
Removed in https://github.com/guardian/members-data-api/pull/371/commits/5b03a52c8c8b69960b7ea6928b19f9d909632f31 (and similarly in https://github.com/guardian/manage-frontend/commit/3dd81ad407ac94a95d6158ed5e428bb39e2365fc)